service/reducers/entities/files.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {FilesTypes, UsersTypes} from 'service/constants';
+
+function files(state = {}, action) {
+    switch (action.type) {
+    case FilesTypes.RECEIVED_FILES_FOR_POST: {
+        const filesById = action.data.reduce((filesMap, file) => {
+            return {...filesMap,
+                [file.id]: file
+            };
+        }, {});
+        return {...state,
+            ...filesById
+        };
+    }
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function fileIdsByPostId(state = {}, action) {
```

Comment:
Oh, I see. And we don't do that on the webapp since `FileInfos` there are stored by their post id. 

Eventually, we could change this over to just ask the server for the `FileInfos` for the `post.file_ids` instead of for the `post.id`, but that's not possible right now because some legacy posts still don't have that field set and need the `getFileInfosForPost` API call to do some migration on them.